modules/integration/src/test/java/io/ballerina/messaging/broker/integration/standalone/jms/DurableTopicMessagesOrderTest.java

```diff
@@ -0,0 +1,270 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+package io.ballerina.messaging.broker.integration.standalone.jms;
+
+import io.ballerina.messaging.broker.integration.util.ClientHelper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testng.Assert;
+import org.testng.annotations.Parameters;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+import javax.jms.JMSException;
+import javax.jms.Session;
+import javax.jms.TextMessage;
+import javax.jms.Topic;
+import javax.jms.TopicConnection;
+import javax.jms.TopicConnectionFactory;
+import javax.jms.TopicPublisher;
+import javax.jms.TopicSession;
+import javax.jms.TopicSubscriber;
+import javax.naming.InitialContext;
+
+/**
+ * Test class to validate message ordering in the durable topic.
+ */
+public class DurableTopicMessagesOrderTest {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DurableTopicMessagesOrderTest.class);
+
+    @Parameters({"broker-port"})
+    @Test(enabled = false)
+    public void test1966DurableTopicMessagesOrderSingleSubscriber(String port) throws Exception {
+        String topicName = "test1966DurableTopicMessagesOrderSingleSubscriber";
+        List<String> subscriberOneMessages = new ArrayList<>();
+        int numberOfMessages = 1966;
+
+        InitialContext initialContext = ClientHelper
+                .getInitialContextBuilder("admin", "admin", "localhost", port)
```

Comment:
Shall we also take these from the testng parameters?

Reply:
Fixed in 5c14195